Platform/OpenVariableRuntimeDxe/VariableDxe.c

```diff
@@ -684,8 +686,18 @@ VariableServiceInitialize (
   //
   // Avoid setting UEFI 2.x interface member on EFI 1.x.
   //
-  if (SystemTable->RuntimeServices->Hdr.Revision >= EFI_2_00_SYSTEM_TABLE_REVISION) {
-    SystemTable->RuntimeServices->QueryVariableInfo = VariableServiceQueryVariableInfo;
+  // First test all systable elements as some may have been spoofed and pass a limited element check
+  // Then check that QueryVariableInfo is specifically available before setting the interface member
+  //
+  if (  ((SystemTable->Hdr.Revision >> 16U) > 1)
+     && ((SystemTable->BootServices->Hdr.Revision >> 16U) > 1)
+     && ((SystemTable->RuntimeServices->Hdr.Revision >> 16U) > 1))
+  {
+    OffsetQVI = OFFSET_OF (EFI_RUNTIME_SERVICES, QueryVariableInfo);
+    HeaderQVI = OffsetQVI + sizeof (SystemTable->RuntimeServices->QueryVariableInfo);
+    if (SystemTable->RuntimeServices->Hdr.HeaderSize >= HeaderQVI) {
+      SystemTable->RuntimeServices->QueryVariableInfo = VariableServiceQueryVariableInfo;
+    }
```

Comment:
The define should work as well I suppose.
On the other checks,, it is for completeness.
It may be decided of course that such completeness is not wanted under any circumstance.

Reply:
Based also on the other discussion in https://github.com/acidanthera/OpenCorePkg/pull/414/files (search for 'gST->Hdr.Revision') I am going to revert these changes and add a comment to the code.